docs/CodeTransformation.md

```diff
@@ -22,109 +22,114 @@ If you override the `transform` configuration option `babel-jest` will no longer
 You can write your own transformer. The API of a transformer is as follows:
 
 ```ts
-interface SyncTransformer<OptionType = unknown> {
-  /**
-   * Indicates if the transformer is capable of instrumenting the code for code coverage.
-   *
-   * If V8 coverage is _not_ active, and this is `true`, Jest will assume the code is instrumented.
-   * If V8 coverage is _not_ active, and this is `false`. Jest will instrument the code returned by this transformer using Babel.
-   */
+// This version of the interface you are seeing on the website has been trimmed down for brevity
+// For the full definition, see `packages/jest-transform/src/types.ts` in https://github.com/facebook/jest 
+// (taking care in choosing the right tag/commit for your version of Jest)
+
+export interface CallerTransformOptions {
+  supportsDynamicImport: boolean;
+  supportsExportNamespaceFrom: boolean;
+  supportsStaticESM: boolean;
+  supportsTopLevelAwait: boolean;
+}
+
+export interface ReducedTransformOptions extends CallerTransformOptions {
+  instrument: boolean;
+}
+
+export type StringMap = Map<string, string>;
+
+export interface TransformOptions<OptionType = unknown>
+  extends ReducedTransformOptions {
+  /** a cached file system which is used in jest-runtime - useful to improve performance */
+  cacheFS: StringMap;
+  config: Config.ProjectConfig;
+  /** A stringified version of the configuration - useful in cache busting */
+  configString: string;
+  /** the options passed through Jest's config by the user */
+  transformerConfig: OptionType;
+}
```

Comment:
Can we combine these into a single `interface TransformOptions<OptionType = unknown>`?

Reply:
As in, for the sake of brevity in the docs only? Yes, I could. But if you are asking if we could _generally_ do it like that, I think the answer is "I think we don't want that?", since `CallerTransformOptions` is used in `jest-runtime` and `ReducedTransformOptions` are used lots of places inside of `babel-jest`.

A combined, docs-only version could be something like this:
```javascript
export interface TransformOptions<OptionType = unknown>
  extends ReducedTransformOptions {
  supportsDynamicImport: boolean;
  supportsExportNamespaceFrom: boolean;
  supportsStaticESM: boolean;
  supportsTopLevelAwait: boolean;
  instrument: boolean;
  /** a cached file system which is used in jest-runtime - useful to improve performance */
  cacheFS:  Map<string, string>;
  config: Config.ProjectConfig;
  /** A stringified version of the configuration - useful in cache busting */
  configString: string;
  /** the options passed through Jest's config by the user */
  transformerConfig: OptionType;
}
```

Which one do you mean? 😸

Reply:
>As in, for the sake of brevity in the docs only?

exactly 👍